File: javascript/arrays/reverse_array.js


let arr = [1,2,3,4,5,10,15];

function swap(arr,s,e){

    let temp = arr[s];
    arr[s] = arr[e];
    arr[e] = temp;
}
function reverseArray(arr){

    let start = 0;
    let end = arr.length - 1;

    while(start < end){
        swap(arr, start, end)
        start++;
        end--;
    }

    console.log(arr)
};

reverseArray(arr)






//Time complexity O(N)
//No extra space so space complexity O(1)

File: c++/arrays/reverse_array.cpp

#include <iostream>
using namespace std;

void reverseArray(int arr[], int n)
{

    int s = 0;
    int e = n - 1;

    while (s < e)
    {
        int temp = arr[s];

        arr[s] = arr[e];
        arr[e] = temp;

        s++;
        e--;
    }
}

int main()
{

    int arr[] = {1, 2, 3, 4, 5, 10};

    int n = sizeof(arr) / sizeof(int);

    reverseArray(arr, n);
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    };
    cout << endl;

    return 0;
}

//Time complexity O(N)
//No extra space so space complexity O(1)